url90b6689baa_adsonrocha_PD321_tgz-pJ8-MyFuzzyLiteClassJ8/src/com/fuzzylite/imex/FclImporter.java

```diff
@@ -0,0 +1,613 @@
+        } else if ("HSUM".equals(name)) {
+            className = HamacherSum.class.getSimpleName();
+        }
+        return FactoryManager.instance().snorm().createInstance(className);
+    }
+
+    protected Term extractTerm(String line) {
+        String spacedLine = "";
+        for (char c : line.toCharArray()) {
+            if (c == '(' || c == ')' || c == ',') {
+                spacedLine += " " + c + " ";
+            } else if (c == ':') {
+                spacedLine += " :";
+            } else if (c == '=') {
+                spacedLine += "= ";
+            } else {
+                spacedLine += c;
+            }
+        }
+
+        final int S_KWTERM = 1, S_NAME = 2, S_ASSIGN = 3,
+                S_TERM_CLASS = 4, S_PARAMETERS = 5;
+        int state = S_KWTERM;
+        StringTokenizer tokenizer = new StringTokenizer(spacedLine);
+        String token, name = "", termClass = "";
+        List<String> parameters = new ArrayList<>();
+        while (tokenizer.hasMoreTokens()) {
+            token = tokenizer.nextToken();
+            if (state == S_KWTERM && "TERM".equals(token)) {
+                state = S_NAME;
+                continue;
+            }
+            if (state == S_NAME) {
+                name = token;
+                state = S_ASSIGN;
+                continue;
+            }
+            if (state == S_ASSIGN && ":=".equals(token)) {
+                state = S_TERM_CLASS;
+                continue;
+            }
+            if (state == S_TERM_CLASS) {
+                if (Op.isNumeric(token)) {
+                    termClass = Constant.class.getSimpleName();
+                    parameters.add(token);
+                } else if ("(".equals(token)) {
+                    termClass = Discrete.class.getSimpleName();
+                } else {
+                    termClass = token;
+                }
+                state = S_PARAMETERS;
+                continue;
+            }
+            if (state == S_PARAMETERS) {
+                if (!Function.class.getSimpleName().equals(termClass)
+                        && ("(".equals(token) || ")".equals(token) || ",".equals(token))) {
+                    continue;
+                }
+                if (";".equals(token)) {
+                    break;
+                }
+                parameters.add(token.trim());
+            }
+        }
+
+        if (state <= S_ASSIGN) {
+            throw new RuntimeException("[syntax error] malformed term in line: " + line);
+        }
+
+        try {
+            Term result = FactoryManager.instance().term().createInstance(termClass);
+            result.setName(Op.makeValidId(name));
+            if (result instanceof Function) {
+                result.configure(Op.join(parameters, ""));//remove spaces for formula of function
+            } else {
+                result.configure(Op.join(parameters, " "));
+            }
+
+            return result;
+        } catch (Exception ex) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: An exception is caught and re-thrown within the catch clause without any other operations being performed on the exception, making the catch clause unnecessary.

**Fix**: Perform additional operations on the exception that has been caught. You can wrap a standard exception into a custom exception to add additional information or special handling. When you do that, make sure to set the original exception as the cause. Otherwise, you will lose the stack trace and message of the original exception, which will make it difficult to analyze the event that caused the exception.  
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)